### PR TITLE
Add Flush method

### PR DESCRIPTION
**Pull request by zorkian**
This adds a Flush method to the consumer which can be called to
immediately invoke a flushing of the latest committed offset. This can
be slow and isn't recommended to be called frequently, but if you need
to checkpoint offsets for some reason this can be used.

### CODE REVIEW

**Comment by zorkian** [comment]
cc @DrTall

File: marshal/consumer.go [hunk 1]
```diff
@@ -746,6 +745,51 @@ func (c *Consumer) Commit(msg *Message) error {
...
+		go func() {
+			defer waiter.Done()
+			if err := cl.Flush(); err != nil {
+				errChan <- err
+			}
+		}()
+	}
+
+	// Wait for all flushes to finish
+	waiter.Wait()
```
Comment (DrTall) [inline marshal/consumer.go]: If I've traced this all down correctly, this not waiting forever hinges on the produce timeouts inside https://mirth.corp.dropbox.com/file/server/go/src/github.com/dropbox/marshal/marshal/marshal.go#L326 ? It makes me somewhat nervous but I guess that if `Heartbeat` was running forever regularly we're going to lose all of our claims anyway?

**Comment by DrTall** [comment]
LGTM

File: marshal/consumer.go (hunk 1 above)
Comment (zorkian) [inline marshal/consumer.go]: As discussed offline: Agreed that this has the potential for never returning but I think that is technically valid for a flush/fsync operation. If the caller cares about timeouts they can add one on their call.

**Pull request merged**